app/app/src/main/kotlin/com/hedvig/app/feature/insurance/ui/tab/InsuranceViewModel.kt

```diff
@@ -24,25 +31,41 @@ class InsuranceViewModel(
   private val hAnalytics: HAnalytics,
 ) : ViewModel() {
 
-  private val _uiState = MutableStateFlow(InsuranceUiState(loading = true))
-  val uiState = _uiState.asStateFlow()
+  private val retryChannel = RetryChannel()
+  private val storeUrl = MutableStateFlow<Uri?>(null)
+  private val loading = MutableStateFlow(false)
+  private val dataOrNullFlow: Flow<List<InsuranceModel>?> = retryChannel.transformLatest {
+    loading.update { true }
+    either {
+      val contracts = getContractsUseCase.invoke().bind()
+      val crossSells = getCrossSellsUseCase.invoke().bind()
+      createInsuranceItems(contracts, crossSells)
+    }.fold(
+      ifLeft = {
+        emit(null)
```

Comment:
Are we logging this error somewhere?

Reply:
Hmm no we don't, and we didn't use to either https://github.com/HedvigInsurance/android/blob/77eacb4f93942ec861ae691b4e92d4bf8f252572/app/app/src/main/kotlin/com/hedvig/app/feature/insurance/ui/tab/InsuranceViewModel.kt#L39
I can add something though for sure.

Right now when this happens, they get the error screen (since `hasError` evaluates to true further down where we do `hasError = insuranceModels == null && loading == false,`), and they can always retry loading by clicking the button. So it's an easily recoverable error if you were worried about that part.